app.py

```diff
@@ -44,4 +44,176 @@ def show_breakdown_dialog(self):
         self.engine.execute_in_main_thread(fn)
         
         
+    def analyze_scene(self):
+        """
+        Runs the scene analysis and returns a list of scene items.
+        
+        Each item is represented by a dictionary with a number of keys to 
+        describe the item. This method uses the same logic that the main UI
+        uses to determine the list of files.
+        
+        Only files whose path corresponds to a template in the toolkit templates
+        file will be detected. Files do not need to exist as publishes in Shotgun
+        however if they do, this method will return basic Shotgun publish metadata
+        for them.
+        
+        The two keys node_name and node_type are used to return a DCC-centric 
+        "address" or representation which makes it possible to identify the path
+        within the DCC. In for example Maya and Nuke, this will return the 
+        node name and type. The logic for this is implemented in the hooks and 
+        will vary between DCCs.
+        
+        Here is an example of what the return data typically looks like:
+        
+        {'fields': {'Sequence': 'aaa',
```

Comment:
I wonder if the app API needs to return all this data - it really feels like it's exposing a lot of the internal functionality to the caller and also makes them understand this data before they can use the other methods.  Would this be too simple:

```python
# get a list of items (each item in the list is dictionary containing {'node type', 'node name', 'path', 'version'}
items = app.analyse_scene()

for item in items:
    # calculate the latest version:
    latest_version = app.calculate_latest_version(item["path"])

    # if latest is newer then update:
    if item["version"] < latest_version:
        app.update_item(item["node_type"], item["node_name"], item["path"], latest_version)
```
?